### PR TITLE
Changes to run STL Trex on Elba

### PR DESCRIPTION
**Pull request by nmram**
Changes to rudra app are in  https://github.com/pensando/sw/commit/21653d240b64717a41fe9069f8f2b723b57c6109

### CODE REVIEW

File: src/drivers/trex_driver_ionic.cpp [hunk 1]
```diff
@@ -159,3 +162,66 @@ void CTRexExtendedDriverBaseIonic::get_rx_stat_capabilities(uint16_t &flags, uin
     base_ip_id = IP_ID_RESERVE_BASE;
 }
 
+void CTRexExtendedDriverBaseIonic::add_del_rules(enum trex_rte_filter_op op, repid_t  repid, uint32_t type,
+                                                 uint32_t type_mask,uint8_t ttl, uint8_t ttl_mask,
+                                                 uint32_t id, uint32_t id_mask, int queue) {
+
+    if (op == TREX_RTE_ETH_FILTER_ADD) {
```
Comment (akshaydorwat) [inline src/drivers/trex_driver_ionic.cpp]: ionic dev calls are supported only on the Rudra modified classic app. Hence , invoke them only under ENV variable  `RUDRA_HW_OFFLOAD`

File: src/drivers/trex_driver_ionic.cpp [hunk 2]
```diff
@@ -159,3 +162,66 @@ void CTRexExtendedDriverBaseIonic::get_rx_stat_capabilities(uint16_t &flags, uin
...
+
+int CTRexExtendedDriverBaseIonic::configure_rx_filter_rules(CPhyEthIF * _if) {
+    return configure_rx_filter_rules_internal(_if, TREX_RTE_ETH_FILTER_ADD);
+}
+
+int CTRexExtendedDriverBaseIonic::configure_rx_filter_rules_internal(CPhyEthIF * _if,
+                                                                     enum trex_rte_filter_op op) {
+    repid_t repid=_if->get_repid();
+
+    if (get_is_stateless()) {
```
Comment (akshaydorwat) [inline src/drivers/trex_driver_ionic.cpp]: same as above

File: src/drivers/trex_driver_ionic.cpp (hunk 1 above)
Comment (nmram) [reply src/drivers/trex_driver_ionic.cpp]: Done

**Review by akshaydorwat — APPROVED** [review]
LGTM

**Pull request merged**